engine/plugins/server/server.js

```diff
@@ -66,28 +91,83 @@ module.exports = function(options, imports, register) {
             socket.emit('UpdateState', client.state);
         });
 
+        /**
+         * The first message the client should send is "ClientReady" which will validate the token and send
+         * configuration to the client base on that token.
+         */
         socket.on('ClientReady', (data) => {
-            // @TODO: Make sure only one socket is open pr. client.
-            // @TODO: Validate token.
-
-            // Emit event to state machine.
-            bus.emit('state_machine.start', {
-                token: data.token,
-                busEvent: busEvent
+            token = data.token;
+            fetch(options.tokenEndPoint + token).then(res => res.json()).then(data => {
+                // Validate the token and send error if not valid.
+                if (Object.prototype.hasOwnProperty.call(data, 'valid') && !data.valid) {
+                    socket.emit('error', { message: 'Not authorized', code: 401 });
+                    socket.disconnect(true);
+                    return;
+                }
+
+                // Set that current token is valid.
+                isTokenValid = true;
+
+                // Get configuration for this client box based on config id from token validation.
+                const clientEvent = uniqId();
+                bus.on(clientEvent, (config) => {
```

Comment:
Is this on(clientEvent) cleaned up later? E.g. in the on("disconnect") ...

Reply:
Fixed